File: test/kalai/demo/demo01.clj
(ns kalai.demo.demo01
  (:refer-clojure :exclude [format])
  (:require [kalai.common :refer :all]))

(defclass "NumFmt"
  (defn format ^String [^Integer num]
    (let [^Integer i (atom num)
          ^StringBuffer result (atom (new-strbuf))]
      (while (not (= @i 0))
        (let [^Integer quotient (quot @i 10)
              ^Integer remainder (rem @i 10)]
          (prepend-strbuf @result remainder)
          (reset! i quotient)))
      (return (tostring-strbuf @result)))))

File: examples/rust/src/a/demo_01.rs
use crate::kalai::kalai;
use crate::kalai::kalai::PMap;
pub fn format(num: i32) -> String {
    let mut i: i32 = num;
    let mut result: std::vec::Vec<char> = std::vec::Vec::new();
    {
        while !(i == 0i32) {
            let quotient: i32 = (i / 10i32);
            let remainder: i32 = (i % 10i32);
            {
                result.splice(
                    0..0,
                    remainder
                        .to_string()
                        .chars()
                        .collect::<std::vec::Vec<char>>(),
                );
                i = quotient;
            }
        }
        return result.into_iter().collect();
    }
}
pub fn main() {
    let _args: std::vec::Vec<String> = std::env::args().collect();
    {
        format(2345i32);
        println!("{}", format(2345i32));
    }
}
